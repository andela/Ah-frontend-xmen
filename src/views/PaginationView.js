/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Pagination from '../components/Pagination/Pagination';
import fetchArticlesAction from '../actions/articleActions/fetchArticlesAction';

export class Paginations extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageNumbers: [],
      allArticles: 7,
      currentPage: 1,
      totalPages: 1,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.count && nextProps.count > 1) {
      if (nextProps.count > 6) {
        this.setState({
          allArticles: nextProps.count,
          totalPages: nextProps.count / 6,
        });
      } else {
        this.setState({ allArticles: nextProps.count, totalPages: 1 });
      }
    }
  }

  handleNext = (nextPageIndex) => {
    if (nextPageIndex < +this.state.totalPages && nextPageIndex >= 1) {
      this.props.fetchArticlesAction(nextPageIndex);
      this.setState({ currentPage: nextPageIndex });
    }
  };

  handlePrevious = () => {
    this.props.fetchArticlesAction(this.state.currentPage - 1);
  };

  render() {
    const { currentPage, totalPages } = this.state;
    let { pageNumbers } = this.state;
    pageNumbers = [];

    for (let i = 1; i <= totalPages; i += 1) {
      pageNumbers.push(i);
    }
    return (
      <Pagination
        pageNumbers={pageNumbers}
        nextPage={this.handleNext}
        previousPage={this.handlePrevious}
        currentPage={currentPage}
      />
    );
  }
}

export const mapStateToProps = state => ({
  newArticles: state.articles.articles,
  count: state.articles.allArticles,
});

export default connect(
  mapStateToProps,
  {
    fetchArticlesAction,
  },
)(Paginations);
