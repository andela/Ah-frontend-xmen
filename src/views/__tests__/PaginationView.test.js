import React from 'react';
import { shallow } from 'enzyme';
import { Paginations } from '../PaginationView';

const props = {
  count: 103,
  newArticles: [],
  fetchArticlesAction: jest.fn(),
};

describe('<Paginations />', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Paginations {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Test that the state', () => {
  it('should update in componentwillrecieveprops', () => {
    const wrapper = shallow(<Paginations />);
    const nextProps = {
      count: 5,
    };
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state());
  });

  it('should update in handlenext', () => {
    const wrapper = shallow(<Paginations {...props} />);
    wrapper.instance().handleNext(0);
    expect(wrapper.state());
  });
});
