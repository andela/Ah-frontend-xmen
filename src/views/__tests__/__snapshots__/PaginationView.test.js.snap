// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginations /> should render without crashing 1`] = `ShallowWrapper {}`;
